juju_verify/verifiers/base.py

```diff
@@ -46,19 +46,26 @@ class BaseVerifier:
 
     NAME = ""
 
-    def __init__(self, units: List[Unit]):
+    def __init__(
+        self, units: List[Unit], exclude_affected_units: Optional[List[Unit]] = None
+    ):
         """Initiate verifier linked to the Juju units.
 
         All the checks that the verifier implements must expect that the action
         that is being verified is intended to be performed on all juju units
         in the 'self.units' simultaneously.
 
+        Parameter `exclude_affected_units` should contain list of units that are
+        explicitly checked by other Verifier(s) so that they can be excluded from
+        the warnings in `check_affected_machines()`.
+
         :raises VerificationError: If 'units' parameter is empty
         :raises VerificationError: If 'units' parameter contains units from
                                    different models.
         """
         self.units = units
         self.affected_machines = set()
+        self.exclude_affected_units = exclude_affected_units or []
```

Comment:
What do you say, if the `exclude_affected_units` is provided as list of ids? I don't see any use case as a Unit object and it will always transform into an ids list.

Reply:
And also `self.affected_machines` is defined as `set`.

Reply:
I'd like to keep this as `List[Unit]` to keep it consistent with `self.units` (it'll also keep `__init__` arguments consistent). One way or another, it'll be different from one of the other properties. However I don't see the difference of types as a problem.

Reply:
IMO, this variable will be used `entity_id in self.exclude_affected_units`, so you will always need to transfer this to list of ids or similar approach.